### ISSUE TITLE
Variables declared in the Properties of the Lua Script component are treated as nil until their exposed parameter is changed on the component

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#2262]
**Describe the bug**
It has been noticed, that the variables declared in the Properties of the Lua Script are not properly processed, until the variable is changed in the Lua Script component.

**To Reproduce**
1. Create an entity with Lua Script component attached.
2. Add a script to the Lua Script component and copy/paste the code provided below to it.
3. Enter the Game Mode. 

**Expected behavior**
The variable is properly used, without having to change it on the Lua Script component first.

**Video and Lua script**

https://user-images.githubusercontent.com/86952082/126158698-71dfc785-109a-4cc9-930e-080a40db63a6.mp4

```
--Please name the script Repro.lua when creating it for this script to be compatible.
local Repro =
{
	Properties =
	{
		myVar = "ABCD",
	}
}

function Repro:OnActivate()
	Debug.Log(tostring(self.Properties.myVar));
end

return Repro;
```

**Desktop**

- Device: PC
- OS: Windows
- Version 10
- CPU AMD Ryzen 5 3600 6-Core Processor, 3600 MHz, 6 Core(s), 12 Logical Processor(s)
- GPU NVIDIA GeForce RTX 2060 SUPER
- Memory 16GB

**Comment by ppinfel** [comment]
Any updates 👀

**Comment by ppinfel** [comment]
Ok random info its still here on latest dev

**Comment by carlitosan** [comment]
taking a look now

**Comment by carlitosan** [comment]
The issues goes away after saving the level and restarting the editor.

**Comment by carlitosan** [comment]
Fix going up shortly.

**Comment by carlitosan** [comment]
closed by: https://github.com/o3de/o3de/pull/7369

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (b1df4c9) build.
